edx_repo_tools/codemods/django3/github_actions_modernizer_django.py

```diff
@@ -0,0 +1,72 @@
+"""
+Django Matrix Modernizer for Github Actions CI
+"""
+import re
+from copy import deepcopy
+
+import click
+
+from edx_repo_tools.utils import YamlLoader
+
+DJANGO_PATTERN = r"django[0-3][0-2][0-2]?"
+DJANGO_VERSIONS = ['2.2', '3.0', '3.1', '3.2']
+ALLOWED_DJANGO_ENVS = ['django22', 'django30', 'django31', 'django32']
+
+
+class GithubCIDjangoModernizer(YamlLoader):
+    def __init__(self, file_path):
+        super().__init__(file_path)
+        self.yml_instance.default_flow_style = None
+        self.yml_instance.indent(mapping=2, sequence=2, offset=0)
+
+    def _update_matrix(self):
+
+        for section_key in self.elements['jobs'].keys():
+            matrix_keys = self.elements['jobs'][section_key]['strategy']['matrix'].keys()
+
+            django_keys = [key for key in matrix_keys if 'django' in key]
```

Comment:
we should instead be filtering out matrix items based on values not on keys, let's say if a file has matrix keys
```django: ['django22']``` and another file has ```toxenv: ['django22', 'quality']```
a filter based on values should update both values for both of these where matrix item is a Django version, then we shouldn't need another iteration over matrix items to find `tox` key like we're doing now

Reply:
The current implementation will handle the django environments for both cases like `django: ['2.2', '3.0']` and `django-version: ['2.2', '3.0']` and will update the `toxenv` separately because we have different env structure for the django versions i.e. `['2.2', '3.0`]` for `django-versions` and `['django22', 'django30']` for `toxenv`. 
Changing the current implementation will need the whole functionality to be re-written to handle both cases in a single iteration but I believe the current implementation is more suitable for our requirements since we have `django-versions` key pattern being followed in almost all the ci files as well.